junifer/datagrabber/datalad_base.py

```diff
@@ -109,28 +110,59 @@ def _dataset_get(self, out: Dict) -> Dict:
                 # was to point to a subdataset rather than a file. This may be
                 # a source of confusion (+ performance/storage issue) when
                 # implementing a grabber.
-                self._dataset.get(v["path"])
+                dl_out = self._dataset.get(v["path"])
+                if not self._was_cloned:
+                    # If the dataset was already installed, check that the
+                    # file was actually downloaded to avoid removing a
+                    # file that was already there.
+                    if len(dl_out) > 1:
+                        raise_error("More than one file downloaded. WTF?")
+                    dl_out = dl_out[0]
+                    if dl_out["status"] == "ok":
+                        logger.debug("File downloaded")
+                        self._got_files.append(v["path"])
+                    elif dl_out["status"] == "notneeded":
+                        dirty_status = True
+                        logger.debug("File was already present")
+                    else:
+                        raise_error(f"File download failed: {dl_out}")
                 logger.debug("Get done")
 
+        assert self._dataset.repo is not None  # avoid errors from mypy
```

Comment:
Maybe raise as error here?

Reply:
No need, this is just because mypy can't understand that `_dataset.repo` is not None at this point.

Reply:
I'm a bit unsure of adding an `assert` statement just because of `mypy`.

Reply:
well, it's also good to double check

Reply:
Yeah that's why I thought an error would be better for this case.